src/plugins/intel_cpu/src/nodes/deconv.cpp

```diff
@@ -349,16 +349,46 @@ std::pair<VectorDims, VectorDims> Deconvolution::makeDummyInOutShape() {
                                                                                               std::max(minDims[i + 2], static_cast<Dim>(64))) : dims[i + 2];
                 }
             }
-            ov::CoordinateDiff pb = autoPad ? ov::CoordinateDiff(paddingL.size(), 0) : paddingL;
-            ov::CoordinateDiff pe = autoPad ? ov::CoordinateDiff(paddingR.size(), 0) : paddingR;
 
             const auto& origInDims = getInputShapeAtPort(0).getDims();
+            const auto& origInMinDims = getInputShapeAtPort(0).getMinDims();
+            const auto& origInMaxDims = getInputShapeAtPort(0).getMaxDims();
             const auto& weightDims = getWeightDims();
             const size_t wghOffset = getAlgorithm() == Algorithm::DeconvolutionGrouped ? 1 : 0;
+
+            const auto paddings = [&]() {
+                VectorDims paddings(paddingL.size());
+                if (!autoPad) {
+                    for (size_t i = 0; i < paddings.size(); ++i) {
+                        paddings[i] = paddingL[i] + paddingR[i];
+                    }
+                } else {
+                    for (size_t i = 0; i < origInDims.size() - 2; i++) {
+                        if (origInDims[i + 2] == Shape::UNDEFINED_DIM &&
+                            (origInMinDims[i + 2] != 0 || origInMaxDims[i + 2] != Shape::UNDEFINED_DIM)) {
+                            // if input shape is dynamic and boundable, paddings should be computed basing on the following limitations:
+                            // 1. paddings must be non negative
+                            // 2. paddings must be between min and max posible values of the corresponding input dimension
+                            auto c1 = lastOutputSpatialDims[i] - outputPadding[i] - 1 -
+                                      (dilation[i] + 1) * static_cast<int32_t>(weightDims[wghOffset + 2 + i] - 1);
+                            auto upper_bound = stride[i] * static_cast<int32_t>(origInMaxDims[i + 2] - 1) - c1;
+                            if (upper_bound < 0) {
+                                IE_THROW() << errorPrefix << ": paddings for dummy shapes can't be computed";
+                            }
+                            auto lower_bound = stride[i] * static_cast<int32_t>(origInMinDims[i + 2] - 1) - c1;
+                            if (lower_bound > 0) {
+                                paddings[i] = lower_bound;
+                            }
+                        }
+                    }
+                }
+                return paddings;
+            }();
```

Comment:
It seems that it does not make sense to create this lambda. Since it is called immediately and is not used anywhere else. Suggest inline the lambda body right here.

Reply:
Done, thanks